DOCSUP-791: Docs for optimize_read_in_order setting (#12611)
ClickHouse/ClickHouse: docs/en/sql-reference/statements/select/order-by.md

External sorting works much less effectively than sorting in RAM.

> ## Optimize_read_in_order Setting {#optimize_read_in_order}

Reviewer: ```suggestion
## Optimization of Data Reading {#optimize_read_in_order}
```

>  If `ORDER BY` expression has coinciding prefix with the table sorting key, you can optimize the query by using the [optimize_read_in_order](../../../operations/settings/settings.md#optimize_read_in_order) setting.  

Reviewer: ```suggestion
 If `ORDER BY` expression has a prefix that coincides with the table sorting key, you can optimize the query by using the [optimize_read_in_order](../../../operations/settings/settings.md#optimize_read_in_order) setting.  
```

 
 When the `optimize_read_in_order` setting is enabled, the Clickhouse server uses the table index and reads the data in order of the `ORDER BY` key. This allows to avoid reading all data in case of specified [LIMIT](../../../sql-reference/statements/select/limit.md). So queries on big data with small limit are processed faster.

Optimization works with both `ASC` and `DESC` and doesn't work together with [GROUP BY](../../../sql-reference/statements/select/group-by.md) clause and [FINAL](../../../sql-reference/statements/select/from.md#select-from-final) modifier.

When the `optimize_read_in_order` setting is disabled, the Clickhouse server does not use the table index while processing `SELECT` queries.

> There are queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found. For these queries you should consider disabling `optimize_read_in_order` manually.

Reviewer: ```suggestion
Consider disabling `optimize_read_in_order` manually, when running queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found.
```

> Optimization is supported in [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md) engine, and also in [Merge](../../../engines/table-engines/special/merge.md), [Buffer](../../../engines/table-engines/special/buffer.md) and [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables.

Reviewer: ```suggestion
Optimization is supported in the following table engines:

- [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md)
- [Merge](../../../engines/table-engines/special/merge.md)
- [Buffer](../../../engines/table-engines/special/buffer.md) 
- [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables
```

> For `MaterializedView` storage the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.

Reviewer: ```suggestion
In `MaterializedView`-egine tables the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.
```

Optimization works with both `ASC` and `DESC` and doesn't work together with [GROUP BY](../../../sql-reference/statements/select/group-by.md) clause and [FINAL](../../../sql-reference/statements/select/from.md#select-from-final) modifier.

When the `optimize_read_in_order` setting is disabled, the Clickhouse server does not use the table index while processing `SELECT` queries.

Consider disabling `optimize_read_in_order` manually, when running queries that have `ORDER BY` clause, large `LIMIT` and [WHERE](../../../sql-reference/statements/select/where.md) condition that requires to read huge amount of records before queried data is found.

Optimization is supported in the following table engines:

- [MergeTree](../../../engines/table-engines/mergetree-family/mergetree.md)
- [Merge](../../../engines/table-engines/special/merge.md)
- [Buffer](../../../engines/table-engines/special/buffer.md) 
- [MaterializedView](../../../engines/table-engines/special/materializedview.md) storages with underlying `MergeTree` tables

> In `MaterializedView`-egine tables the optimization works with saved queries like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the saved query doesn't have `ORDER BY` clause.

Reviewer 2: ```suggestion
In `MaterializedView`-engine tables the optimization works with views like `SELECT ... FROM merge_tree_table ORDER BY pk`. But it is not supported in the queries like `SELECT ... FROM view ORDER BY pk` if the view query doesn't have the `ORDER BY` clause.
```
